src/common/utils/deploy.utils.ts

```diff
@@ -58,13 +59,36 @@ export default class DeployUtils {
     return flags;
   }
 
-  static getComponentSecrets(individual_secrets: string[], secrets_file?: string): Dictionary<Dictionary<string | number | null>> {
-    const component_secrets = DeployUtils.readSecretsFile(secrets_file);
+  static getAllSecretFiles(raw_argv: any): string[] {
+    const all_secret_files = [];
+    for (const raw_flag of raw_argv) {
+      if (raw_flag.type === 'flag' && (raw_flag.flag === 'secret-file' || raw_flag.flag === 'values')) {
+        all_secret_files.push(raw_flag.input);
+      }
+    }
+    return all_secret_files;
+  }
+
+  static getComponentSecrets(individual_secrets: string[], secrets_file?: string | string[]): Dictionary<Dictionary<string | number | null>> {
+    // Check to see if there are multiple secret files; else, just read the single secret file
+    let component_secrets: any = {};
+    if (Array.isArray(secrets_file)) {
+      for (const secret_file of secrets_file) {
```

Comment:
With the changes above this should become an array by default and the check unneeded. We should just assume that secrets file is an array no matter the size and work with it that way.